src/reducers/entities/helpers.js

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) 2016 Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+function isPostListNull(pl) {
+    if (!pl) {
+        return true;
+    }
+
+    if (!pl.posts) {
+        return true;
+    }
+
+    return !pl.order;
+}
+
+function makePostListNonNull(pl) {
+    let postList = pl;
+    if (!postList) {
+        postList = {order: [], posts: {}};
+    }
+
+    if (!postList.order) {
+        postList.order = [];
+    }
+
+    if (!postList.posts) {
+        postList.posts = {};
+    }
+
+    return postList;
+}
+
+export function addPosts(state, action) {
+    const newState = {...state};
```

Comment:
cool I'll write some unit tests with deepFreeze for this function to make sure its not mutating anything